packer/builder/azure/driver_restapi/request/RetryPolicy.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package request
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	"github.com/MSOpenTech/packer-azure/packer/builder/azure/driver_restapi/response/model"
+)
+
+func defaultRetryPolicy() retryPolicy {
+	return retryPolicy{
+		exponentialBackoff("Throttling", func(err model.AzureError) bool {
+			return err.Code == "TooManyRequests"
+		}, 5*time.Second, 2*time.Minute, 0),
+		constantBackoff("MissingOrIncorrectVersionHeader", func(err model.AzureError) bool {
+			return err.Code == "MissingOrIncorrectVersionHeader" // shouldn't occur, since we always add it
+		}, 0, 100),
+		constantBackoff("InternalError", func(err model.AzureError) bool {
+			return err.Code == "InternalError"
+		}, 10*time.Second, 100),
+		constantBackoff("Conflict/InUse", func(err model.AzureError) bool {
+			return (err.Code == "BadRequest" && strings.Contains(err.Message, "is currently in use by")) ||
+				(err.Code == "ConflictError" && strings.Contains(err.Message, "that requires exclusive access"))
+		}, 10*time.Second, 100),
+	}
+}
+
+type retryRule func(model.AzureError) (bool, time.Duration)
+type retryPolicy []retryRule
+
+func (rules *retryPolicy) ShouldRetry(err model.AzureError) (shouldRetry bool, backoff time.Duration) {
+	for _, rule := range []retryRule(*rules) {
+		shouldRetry, backoff = rule(err)
+		if shouldRetry {
+			return
+		}
+	}
+	return false, 0
+}
+
+func constantBackoff(name string, match func(model.AzureError) bool, backoff time.Duration, maxRetries int) retryRule {
```

Comment:
`type matchRule func(model.AzureError) bool ` could be useful as it also repeats a lot